flink-formats/flink-avro-glue-schema-registry/src/main/java/org/apache/flink/formats/avro/glue/schema/registry/GluSchemaRegistryAvroFormatFactory.java

```diff
@@ -0,0 +1,148 @@
+import static org.apache.flink.formats.avro.glue.schema.registry.GlueSchemaRegistryAvroOptions.CACHE_TTL_MS;
+import static org.apache.flink.formats.avro.glue.schema.registry.GlueSchemaRegistryAvroOptions.COMPATIBILITY;
+import static org.apache.flink.formats.avro.glue.schema.registry.GlueSchemaRegistryAvroOptions.COMPRESSION_TYPE;
+import static org.apache.flink.formats.avro.glue.schema.registry.GlueSchemaRegistryAvroOptions.ENDPOINT;
+import static org.apache.flink.formats.avro.glue.schema.registry.GlueSchemaRegistryAvroOptions.RECORD_TYPE;
+import static org.apache.flink.formats.avro.glue.schema.registry.GlueSchemaRegistryAvroOptions.REGISTRY_NAME;
+import static org.apache.flink.formats.avro.glue.schema.registry.GlueSchemaRegistryAvroOptions.SCHEMA_REGISTRY_SUBJECT;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.amazonaws.services.schemaregistry.utils.AWSSchemaRegistryConstants;
+
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.formats.avro.AvroRowDataDeserializationSchema;
+import org.apache.flink.formats.avro.AvroRowDataSerializationSchema;
+import org.apache.flink.formats.avro.AvroToRowDataConverters;
+import org.apache.flink.formats.avro.RowDataToAvroConverters;
+import org.apache.flink.formats.avro.typeutils.AvroSchemaConverter;
+import org.apache.flink.table.connector.ChangelogMode;
+import org.apache.flink.table.connector.format.DecodingFormat;
+import org.apache.flink.table.connector.format.EncodingFormat;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DeserializationFormatFactory;
+import org.apache.flink.table.factories.DynamicTableFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.factories.SerializationFormatFactory;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.RowType;
+
+/**
+ * Table format factory for providing configured instances of AWS Glue Schema
+ * Registry Avro to RowData {@link SerializationSchema} and
+ * {@link DeserializationSchema}.
+ */
+public class GluSchemaRegistryAvroFormatFactory implements DeserializationFormatFactory, SerializationFormatFactory {
+    public static final String IDENTIFIER = "avro-glue";
+
+    @Override
+    public DecodingFormat<DeserializationSchema<RowData>> createDecodingFormat(DynamicTableFactory.Context context,
+            ReadableConfig formatOptions) {
+        FactoryUtil.validateFactoryOptions(this, formatOptions);
+        var configMap = buildConfigMap(formatOptions);
```

Comment:
We're on Java 8 and cannot use `var`. Please use `final Map<…>`.

Reply:
fixed.  Will also remove `Set.of(...)`